### DISCUSSION TITLE
missing underlines when markdown source file name contains underlines

### DISCUSSION THREAD

**Discussion by xie-dd** [mkdocs/mkdocs#3758] [Q&A] [answered]
My markdown source file name contains underlines, but the sidebar of the site page generated by serve and bulid will have the problem of missing underlines.

![serve](https://github.com/mkdocs/mkdocs/assets/148318644/6afd782f-8e0c-40fc-b867-b7a3a1eecebb)
![site](https://github.com/mkdocs/mkdocs/assets/148318644/c4716622-6a9a-4414-9af9-95b8e0b67cdc)

**Comment by xie-dd** [comment]
```
mkdocs_underline
│  mkdocs.yml
│  serve.jpg
│  site.jpg
│
├─docs
│      index.md
│      test_underline.md
│
└─site
```

**Comment by pawamoy** [comment, marked as answer]
Hi @xie-dd!

This behavior is expected. If you don't specify titles, either through the `nav`, or with a Markdown `# Title`, or with a front-matter `title: Title`, MkDocs uses the file name, within which it replaces underscores with spaces and title-ize the first word.

**Reply by xie-dd** [reply]
Ok, I see. Thank you for your help

**Comment by kamilkrzyskow** [comment]
This was asked 10 days before on the material discussions, and I answered it on the same day @xie-dd:
- https://github.com/squidfunk/mkdocs-material/discussions/7271

**Reply by pawamoy** [reply]
Damn these double discussions are becoming a recurring thing. We should probably document that in issue templates to prevent it 🤔
